src/qililab/instruments/rohde_schwarz/sgs100a.py

```diff
@@ -106,6 +116,12 @@ def set_parameter(self, parameter: Parameter, value: ParameterValue, channel_id:
                 else:
                     self.turn_off()
             return
+        if parameter == Parameter.HARDWARE_MODULATION:
+            value = bool(value)
+            self.settings.hardware_modulation = value
+            if self.is_device_active():
+                self.device.IQ_state(value)
```

Comment:
I would remove some lines and make use of the function hardware_modulation that you have created such as the other parameters above.
```suggestion
            self.settings.hardware_modulation = bool(value)
            if self.is_device_active():
                self.device.IQ_state(self.hardware_modulation)
```